lib/ProductOpener/Data.pm

```diff
@@ -119,6 +122,46 @@ sub execute_count_tags_query ($query) {
 	return execute_tags_query('count', $query);
 }
 
+sub execute_product_query ($parameters_ref, $query, $fields, $sort, $limit, $skip) {
+	# Currently only send descending popularity_key sorts to off-query
+	if ($sort->FETCH('popularity_key') eq -1) {
+		# Convert sort into an array so that the order of keys is not ambiguous
+		my @sort_array = ();
+		if ($sort) {
+			foreach my $k ($sort->Keys) {
+				push(@sort_array, [$k, $sort->FETCH($k)]);
+			}
+		}
+
+		my $results = execute_tags_query(
+			'find',
+			{
+				filter => $query,
+				projection => $fields,
+				sort => \@sort_array,
+				limit => $limit,
+				skip => $skip
+			}
+		);
+
+		if (defined $results) {
```

Comment:
For testing and debugging purposes, it would be great if we could identify if a request was sent to off-query, if we got results, or if it was answered by mongodb. We could record that in $request_ref->{database_request} or something and put it as an HTML comment.

Also it would be interesting to be able to pass a parameter (like ?database=mongodb) to have the query sent to mongodb, so that we can easily compare results.

Reply:
In a sense, it could be a request parameter feature flag. That way we could deploy it without the feature enabled by default, test it with ?database=off-query, and then switch the default.